xarray/core/datatree_mapping.py

```diff
@@ -85,6 +101,12 @@ def map_over_datasets(
 
     from xarray.core.datatree import DataTree
 
+    if kwargs is None:
+        kwargs = {}
+
+    if kwargs:
+        func = partial(func, **kwargs)
```

Comment:
If we use partial I think we can avoid the double `if`. OR could we use `results = func_with_error_context(*node_dataset_args, **kwargs)` to avoid a doubly-wrapped function?

Reply:
Yes, I can change along your suggestion.